packages/enketo-core/src/js/utils.js

```diff
@@ -85,7 +86,16 @@ function getFilename(file, postfix) {
             filenameParts[0] += postfix;
         }
 
-        return filenameParts.join('.');
+        let fileName = filenameParts.join('.');
+
+        // If the fileName contains characters that are not allowed in a URL, encode it.
+        // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURI#description
+        // https://datatracker.ietf.org/doc/html/rfc2396
+        if (!SAFE_FILE_IN_URL_RE.test(fileName)) {
+            fileName = encodeURIComponent(fileName);
```

Comment:
I think you're really better of replacing with `_HASH_` or something... It avoids all the remaining encoding issues that this change creates. This change is supposed to be permanent (it should be imo), but in your other code, it seems like you still do have a lot of decoding/encoding (including in the instance itself(!)) to make it all work.